packages/storybook/src/stories/ino-checkbox/ino-checkbox.stories.ts

```diff
@@ -33,59 +34,51 @@ export default {
       return story();
     },
   ],
-} as Meta;
+  // will be used as default props for all stories
+  args: {
+    checked: false,
+    disabled: false,
+    indeterminate: false,
+    name: '',
+    selection: false,
+    value: '',
+  },
+} as Meta<Components.InoCheckbox>;
 
-export const Playground: Story<Components.InoCheckbox> = (args) => html`
-  <ino-checkbox
-    class="customizable-checkbox"
-    checked="${args.checked}"
-    disabled="${args.disabled}"
-    indeterminate="${args.indeterminate}"
-    name="${args.name}"
-    selection="${args.selection}"
-    value="${args.value}"
-  >
-    Label
-  </ino-checkbox>
-`;
+// the basic template for the checkbox component
+const template = new TemplateGenerator<Components.InoCheckbox>(
+  'ino-checkbox',
+  (args) => html`
+    <ino-checkbox
+      checked="${args.checked}"
+      disabled="${args.disabled}"
+      indeterminate="${args.indeterminate}"
+      name="${args.name}"
+      selection="${args.selection}"
+      value="${args.value}"
+    >
+      Label
+    </ino-checkbox>
+  `
+);
 
-Playground.args = {
-  checked: false,
-  disabled: false,
-  indeterminate: false,
-  name: '',
-  selection: false,
-  value: '',
-};
+export const Playground = template.generateStory({});
 
-export const Variations = (): TemplateResult => html`
-  <div class="story-states">
-    <ino-checkbox checked>Checked</ino-checkbox>
-    <ino-checkbox>Unchecked</ino-checkbox>
-    <ino-checkbox indeterminate>Indeterminated</ino-checkbox>
-  </div>
-  <div class="story-states">
-    <ino-checkbox disabled checked>Checked and Disabled</ino-checkbox>
-    <ino-checkbox disabled>Unchecked and Disabled</ino-checkbox>
-    <ino-checkbox disabled indeterminate
-      >Indeterminate and Disabled
-    </ino-checkbox>
-  </div>
-`;
+export const Checked = template.generateStory({ checked: true }, 'checked');
 
-export const Selection = (): TemplateResult => html`
-  <div class="story-states">
-    <ino-checkbox selection checked>Checked</ino-checkbox>
-    <ino-checkbox selection>Unchecked</ino-checkbox>
-    <ino-checkbox selection indeterminate>Indeterminate</ino-checkbox>
-  </div>
-  <div class="story-states">
-    <ino-checkbox checked selection disabled
-      >Checked and disabled
-    </ino-checkbox>
-    <ino-checkbox selection disabled>Unchecked and disabled</ino-checkbox>
-    <ino-checkbox selection indeterminate disabled>
-      Indeterminate Disabled
-    </ino-checkbox>
-  </div>
-`;
+export const Selection = template.generateStory(
+  {
+    ...Checked.args, // reuse args of Checked Story
```

Comment:
Sjhouldnt this be default and not configurable for all stories that share a template generator? Otherwise I would add abstraction to this to hide the actual implementation (`inheritArguments: bool` or so).